src/test/java/com/easypost/AddressTest.java

```diff
@@ -0,0 +1,145 @@
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @BeforeClass
+    public static void setup() throws EasyPostException{
+        EasyPost.apiKey = "EASYPOST_TEST_API_KEY";
+
+        address = Address.create(Fixture.basicAddress());
+    }
+
+    /**
+     * Test creating an address.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testCreate() throws EasyPostException {
+        Address address = Address.create(Fixture.basicAddress());
+
+        assertTrue(address instanceof Address);
+        assertEquals("388 Townsend St", address.getStreet1());
+        assertTrue(address.getId().startsWith("adr_"));
+    }
+
+    /**
+     * Test creating an address with verify_strict param.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testCreateVerifyStrict() throws EasyPostException {
+        Map<String, Object> addressData = Fixture.basicAddress();
+        List<String> verificationList = new ArrayList<>();
+        
+        verificationList.add("true");
+        addressData.put("verify_strict", verificationList);
+
+        Address address = Address.create(addressData);
+
+        assertTrue(address instanceof Address);
+        assertEquals("388 TOWNSEND ST APT 20", address.getStreet1());
+        assertTrue(address.getId().startsWith("adr_"));
+    }
+
+    /**
+     * Test retrieving an address.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testRetrieve() throws EasyPostException{
+        Address retrievedAddress = Address.retrieve(address.getId());
+
+        assertTrue(retrievedAddress instanceof Address);
+        assertEquals(address.getId(), retrievedAddress.getId());
+    }
+
+    /**
+     * Test retrieving all addresses.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testAll() throws EasyPostException{
+        Map<String, Object> params = new HashMap<>();
+
+        params.put("page_size", Fixture.pageSize());
+
+        AddressCollection addresses = Address.all(params);
+
+        List<Batch> addressesArray = addresses.getAddresses();
+
+        assertTrue(addressesArray.size() <= 5);
+        for(Batch address: addressesArray) {
+            assertTrue(address.getId().startsWith("adr_"));
+        }
+    }
+
+    /**
+     * Test creating a verified address.
+     * We purposefully pass in slightly incorrect data to get the corrected address back once verified.
+     *
+     *  @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testCreateVerify() throws EasyPostException {
+        Address address = Address.create(Fixture.incorrectAddressToVerify());
+
+        assertTrue(address instanceof Address);
+        assertEquals("417 MONTGOMERY ST STE 500", address.getStreet1());
+        assertTrue(address.getId().startsWith("adr_"));
```

Comment:
Let's try and keep all the assertions in the same order (1. instance, 2. ID, 3. property).

This goes for all tests in the library.